src/main/java/io/aiven/kafka/connect/opensearch/RecordConverter.java

```diff
@@ -119,17 +122,44 @@ private DocWriteRequest<?> createDocWriteRequest(final String indexOrDataStreamN
             return docIdStrategy.updateRequest(new DeleteRequest(indexOrDataStreamName), record);
         }
         final var payload = getPayload(record);
-        final var indexRequest = new IndexRequest().index(indexOrDataStreamName);
         if (config.dataStreamEnabled()) {
-            indexRequest
-                    .source(addTimestampToPayload(payload, record.timestamp()), XContentType.JSON)
-                    .opType(DocWriteRequest.OpType.CREATE);
+            return docIdStrategy.updateRequest(
+                    new IndexRequest()
+                            .index(indexOrDataStreamName)
+                            .opType(DocWriteRequest.OpType.CREATE)
+                            .source(addTimestampToPayload(payload, record.timestamp()), XContentType.JSON),
+                    record
+            );
+        } else if (config.indexWriteMethod() == IndexWriteMethod.UPSERT) {
+            if (docIdStrategy == DocumentIDStrategy.NONE) {
```

Comment:
This check has to be done during configuration validation to prevent connector from starting (instead of errorring on each record)